newIDE/app/src/MainFrame/index.js

```diff
@@ -125,6 +126,7 @@ type Props = {
   onChooseProject?: () => Promise<?string>,
   saveDialog?: React.Element<*>,
   onSaveProject?: gdProject => Promise<any>,
+  onSaveAsProject?: gdProject => Promise<any>,
```

Comment:
This is a good idea, though you have to search for all onSaveProject in the codebase and add a "onSaveAsProject" next to it and pass it the proper function.

I help you: go in LocalApp.js and add `onSaveProjectAs={LocalProjectWriter.saveProjectAs}`, next to the line refering to LocalProjectWriter.saveProject
(btw, could you change this from `onSaveAsProject` to `onSaveProjectAs`?)